File: treshold.js
var windowedFrames = []; //2D array - [][]

windowedFrames = [ //Simulated data
    [5, 2, 4, 7, 2, 2], //sum(x^2) = 102
    [8, 5, 4, 7, 8, 4],
    [6, 4, 8, 5, 1, 5],
    [9, 5, 3, 4, 4, 1],
    [6, 4, 2, 4, 4, 1],
]

//__________Initialization output data__________
const out = {
    tresholdEnergy: 0,
    size: 0,
    sum: 0,
    booleanFrameEnergy: 0
}

//__________While loop__________
var whileData = []; //Converted frames to 1D array of their energy
for (let i = 0; i < windowedFrames.length; i++) //windowedFrames = [i][j]
{
    whileData.push(0);
    for (let j = 0; j < windowedFrames[i].length; j++)
        whileData[i] = Math.pow(windowedFrames[i][j], 2) + whileData[i];
}

//__________Sorting array by ASC__________
whileData.sort((b, a) => { return b - a });
//__________Splitting 1D array to two 1D__________
const idx = 14;
var maxIdx = idx;
if (idx > whileData.length) //Limit of size
    var maxIdx = whileData.length;
var firstArray = [];
for (let i = 0; i < maxIdx; i++)
    firstArray.push(whileData[i]);
//__________Estimate mean of firstArray__________
var mean = (firstArray.reduce((a, b) => a + b, 0) / firstArray.length);
//__________Boolean frame energy__________
out.tresholdEnergy = mean * 10;
for (let i = 0; i < whileData.length; i++)
    if (whileData[i] > out.tresholdEnergy)
        out.booleanFrameEnergy = 1; 
        //This is true but there was required convert to int

//__________Finish rest of calculations__________
out.size = whileData.length;
out.sum = whileData.reduce((a,b) => a + b, 0); //sum of all array's elements


console.log(out);

File: treshold.py
import numpy as np
import matplotlib.pyplot as plt
import acquisition as acq
#-------------------------------------------
#            Treshold Detection
#-------------------------------------------

def execute():
    #__________While loop__________
    for i in range(len(acq.data["framesMatrix"])):
        acq.data["whileData"].append(0)
        for j, jVal in enumerate(acq.data["framesMatrix"][i]):
            acq.data["whileData"][i] = pow(acq.data["framesMatrix"][i][j],2) + acq.data["whileData"][i]
    #__________Sorting array by ASC__________
    whileToMean = []
    for item in acq.data["whileData"]:
        whileToMean.append(item)
    whileToMean.sort()
    idx = 14
    maxIdx = idx
    if idx > len(whileToMean):
        maxIdx = len(whileToMean)
    firstArray = []
    for i in range(maxIdx):
        firstArray.append(whileToMean[i])
    #__________Estimate mean of firstArray__________
    mean = sum(firstArray) / len(firstArray)
    #__________Boolean frame energy__________
    acq.data["tresholdEnergy"] = mean * 2000
    for i in range(len(acq.data["whileData"])):
        if acq.data["whileData"][i] > acq.data["tresholdEnergy"]:
            acq.data["booleanFrameEnergy"].append(1)
        else:
            acq.data["booleanFrameEnergy"].append(0)
    #__________Finish rest of calculations__________
    acq.data["size"] = len(acq.data["whileData"])
    acq.data["summary"] = sum(acq.data["whileData"])

    #print(acq.data["out"].__dict__)

def plot():
    #__________Energy Graph__________
    x = np.arange(len(acq.data["whileData"]))
    plt.figure()
    plt.plot(x,acq.data["whileData"])
    plt.plot([0, len(acq.data["whileData"])], [acq.data["tresholdEnergy"], acq.data["tresholdEnergy"]], '--', label='Treshold Energy')
    plt.title("Energy Graph")
    plt.ylabel("Amplitude")
    plt.xlabel("Sample")
    plt.legend()
    plt.show()
    #__________Boolean Energy Graph__________
    x = np.arange(len(acq.data["booleanFrameEnergy"]))
    plt.figure()
    plt.plot(x,acq.data["booleanFrameEnergy"])
    plt.title("Boolean Frame Energy")
    plt.ylabel("Amplitude")
    plt.xlabel("Sample")
    plt.show()